deck-o-cards.js

```diff
@@ -1,49 +1,65 @@
-// 1. Make the function deck_o_cards assemble an array of cards using the provided suits and values arrays.
-// Each card in the deck should be an object formatted as: {suit: 'hearts', value: 'A'}
+
 function deck_o_cards() {
   var values = [2, 3, 4, 5, 6, 7, 8, 9, 10, 'Jack', 'Queen', 'King', 'Ace'];
   var suits = ['hearts', 'diamonds', 'clubs', 'spades'];
 
   var cards = []; // deck
   var shuffledCards = []; // deck shuffled
-
-  // Make 52 card objects and store them in the "cards" array
-  // Hint: use 2 for loops
-
-
-  // 2. Shuffle the cards
-  // Hint: shuffle function is already defined below
-
+  var pokerHand = []; // poker hand
+  var objIndex = 0; //card counter
+  var pokerStatement = "Your poker hand is a(n)"; //string to tell you your poker hand
+
+  // Makes 52 card objects and store them in the "cards" array
+  for (i=0; i < suits.length; i++){
+    for (p=0; p < values.length; p++){
+      cards[objIndex] = {suit: suits[i], value: values[p]};
+      objIndex += 1;
+    };
+  };
+
+  // Shuffles the cards
+  function shuffle(array) {
+      var counter = array.length, temp, index;
+      // While there are elements in the array
+      while (counter > 0) {
+          // Pick a random index
+          index = Math.floor(Math.random() * counter);
+          // Decrease counter by 1
+          counter--;
+          // And swap the last element with it
+          temp = array[counter];
+          array[counter] = array[index];
+          array[index] = temp;
+      }
+      return array;
+  };
+
+  shuffledCards = shuffle(cards);
+
+  //Makes poker hand
+  for (i=0; i < 5; i++){
+    pokerHand[i] = shuffledCards[i];
+  };
+
+  for(i=0; i < pokerHand.length; i++){
+    if (i == pokerHand.length-1){
```

Comment:
Probably want to use strict equality here `===`. Since the whole point of this `if` statement is to simply add "and a(n)" to the list, you could consider using a while loop too.